library/src/main/java/org/apache/apex/malhar/lib/window/accumulation/AbstractPojoJoin.java

```diff
@@ -220,8 +233,24 @@ protected void setObjectForResult(Map<String,PojoUtils.Getter> stream, Object in
         }
         for (Object lObj:left) {
           for (Object rObj:right) {
-            setObjectForResult(leftGettersStream, lObj,o);
-            setObjectForResult(rightGettersStream, rObj,o);
+            if (outputToInputMap != null) {
```

Comment:
Instead of adding a condition here.. If this is null, then the constructor should populate this approapriately and always use it.